### DISCUSSION TITLE
Delphi RADStudio How to manage my units with Git?

### DISCUSSION THREAD

**Discussion by wolfgangbures** [community/community#60831] [New to GitHub]
I have just started with GitHub and barely manged to upload my project to the repository. But how do I best go about loading my units to GitHub? Is there each one in a seperate repository, or can I have one repo for all my units? How will this affect a project?
I admit I have not quite understood the concept... :-/

**Comment by Gustavonn07** [comment]
I don't know if I understood your question correctly, but anything to answer as I understood!

GitHub repositories are typically used to host projects, which consist of multiple files, folders, and units of code (for the project you are using). The organization of units within a repository depends on the structure and complexity of your project.

Here are two common approaches for organizing units within a repository:

Single Repository: You can choose to have one repository for all your units. In this approach, you would organize your units using folders within the repository. Each unit can be contained in its own folder, with the necessary files and dependencies specific to that unit. This approach is suitable when the units are closely related, share common code, or are part of the same project.

Multiple Repositories: Alternatively, you can have separate repositories for each unit. In this approach, each unit would have its own repository with its own set of files and dependencies. This approach is useful when the units are independent of each other or when you want to maintain separate version control and collaboration for each unit.

Deciding which approach to use depends on the nature of your project and how the units relate to each other. If your units are highly interdependent and share common code or resources, a single repository might be more suitable. On the other hand, if the units are independent and can be developed or used separately, separate repositories might be a better choice.

Consider factors such as the size and complexity of the units, the need for collaboration and version control, and the overall organization and maintainability of your project.

Remember, the structure you choose is flexible, and you can always make adjustments as your project evolves. It's important to consider the trade-offs between having everything in one repository or using multiple repositories based on your specific requirements.

In my case, I prefer to create a repository per project or for studies, for example: one for learning javascript, another for sass, another for a project using Tailwind...

I hope this helps clarify how you can organize your units on GitHub. If you have any more questions, feel free to ask!
